src/main/java/dev/sigstore/KeylessVerifier.java

```diff
@@ -0,0 +1,179 @@
+    return new KeylessVerifier.Builder();
+  }
+
+  public static class Builder {
+    private FulcioVerifier fulcioVerifier;
+    private RekorClient rekorClient;
+    private RekorVerifier rekorVerifier;
+
+    public KeylessVerifier.Builder fulcioVerifier(FulcioVerifier fulcioVerifier) {
+      this.fulcioVerifier = fulcioVerifier;
+      return this;
+    }
+
+    public KeylessVerifier.Builder rekorClient(
+        RekorClient rekorClient, RekorVerifier rekorVerifier) {
+      this.rekorClient = rekorClient;
+      this.rekorVerifier = rekorVerifier;
+      return this;
+    }
+
+    public KeylessVerifier build() {
+      Preconditions.checkNotNull(fulcioVerifier);
+      Preconditions.checkNotNull(rekorVerifier);
+      return new KeylessVerifier(fulcioVerifier, rekorClient, rekorVerifier);
+    }
+
+    public Builder sigstorePublicDefaults()
+        throws IOException, InvalidAlgorithmParameterException, CertificateException,
+            InvalidKeySpecException, NoSuchAlgorithmException {
+      var fulcioCert =
+          Resources.toByteArray(
+              Resources.getResource("dev/sigstore/tuf/production/fulcio_v1.crt.pem"));
+      var ctfePublicKey =
+          Resources.toByteArray(Resources.getResource("dev/sigstore/tuf/production/ctfe.pub"));
+      var rekorPublicKey =
+          Resources.toByteArray(Resources.getResource("dev/sigstore/tuf/production/rekor.pub"));
+      fulcioVerifier(FulcioVerifier.newFulcioVerifier(fulcioCert, ctfePublicKey));
+      rekorClient(RekorClient.builder().build(), RekorVerifier.newRekorVerifier(rekorPublicKey));
+      return this;
+    }
+  }
+
+  public KeylessVerificationResult verifyOnline(
+      byte[] artifactDigest, byte[] certChain, byte[] signature)
+      throws IOException, NoSuchAlgorithmException, InvalidKeyException, CertificateException,
+          SignatureException {
+    var signingCert = SigningCertificate.from(Certificates.fromPemChain(certChain));
+    var leafCert = signingCert.getLeafCertificate();
+
+    var checks = KeylessVerificationResult.builder();
+
+    try {
+      fulcioVerifier.verifyCertChain(signingCert);
+      checks.checkPassed(Check.FULCIO_CERT);
+    } catch (FulcioVerificationException ex) {
+      return checks.checkFailed(Check.FULCIO_CERT, ex.getMessage());
+    }
+
+    try {
+      fulcioVerifier.verifySct(signingCert);
+      checks.checkPassed(Check.FULCIO_CERT_SCT);
+    } catch (FulcioVerificationException ex) {
+      return checks.checkFailed(Check.FULCIO_CERT_SCT, ex.getMessage());
+    }
+
+    var hashedRekordRequest =
+        HashedRekordRequest.newHashedRekordRequest(
+            artifactDigest, Certificates.toPemBytes(leafCert), signature);
+    Optional<RekorEntry> rekorEntry;
+    try {
+      rekorEntry = rekorClient.getEntry(hashedRekordRequest);
+    } catch (IOException ex) {
+      return checks.checkFailed(Check.REKOR_ENTRY_EXISTS, "retrieval failure: " + ex.getMessage());
```

Comment:
Technically this isn't about existence check but about a network failure or something.  Maybe just pass on to client?